### PR TITLE
Beta deployment APIs

### PR DESCRIPTION
**Pull request by majolo**
/uncc

### CODE REVIEW

**Comment by majolo** [comment]
Hey @Helcaraxan I've fixed the example code up now so it's ready for review and release once approved.

File: examples/CapacityLimit/Program.cs [hunk 1]
```diff
@@ -130,7 +128,7 @@ protected override void Run()
                 new CreateLoginTokenRequest
                 {
                     PlayerIdentityToken = playerIdentityTokenResponse.PlayerIdentityToken,
-                    DeploymentId = _deployment.Id,
+                    DeploymentId = suitableDeployment.Id.ToString(),
```
Comment (sebzz) [inline examples/CapacityLimit/Program.cs]: 🤕

File: examples/CapacityLimit/Program.cs [hunk 2]
```diff
@@ -97,23 +94,24 @@ protected override void Setup()
...
             Console.WriteLine("Finding current running deployment");
-            _deployment = _deploymentServiceClient
-                .ListDeployments(new ListDeploymentsRequest
-                {
-                    ProjectName = ProjectName,
-                    DeploymentName = DeploymentName
-                })
-                .First(d => d.Status == Deployment.Types.Status.Running);
-            Console.WriteLine($"Found deployment {_deployment.Id}");
+            suitableDeployment = _deploymentServiceClient.GetRunningDeploymentByName(new GetRunningDeploymentByNameRequest
```
Comment (sebzz) [inline examples/CapacityLimit/Program.cs]: `suitableDeployment` -> `runningDeployment` ?

File: examples/CapacityLimit/Program.cs [hunk 3]
```diff
@@ -154,31 +152,33 @@ protected override void Run()
...
+            };
+            setWorkerCapacitiesRequest.WorkerConnectionCapacities.Add(new WorkerCapacity
             {
                 WorkerType = ScenarioWorkerType,
                 MaxCapacity = 3
             });
-            _deploymentServiceClient.UpdateDeployment(new UpdateDeploymentRequest {Deployment = _deployment});
-
+            _deploymentServiceClient.SetDeploymentWorkerCapacities(setWorkerCapacitiesRequest);
+
```
Comment (Helcaraxan) [inline examples/CapacityLimit/Program.cs]: Superfluous whitespace

**Comment by Helcaraxan** [comment]
It appears that this PR is failing when trying to perform the new calls. Are we sure that the new Beta service is implemented by `spatiald`?

File: examples/CapacityLimit/Program.cs (hunk 1 above)
Comment (majolo) [reply examples/CapacityLimit/Program.cs]: What do you mean? Unfortunately this request still takes the deployment id as a string - not much I can do.

**Comment by majolo** [comment]
@Helcaraxan yes that's true, it's not yet implemented but will be soon. Workflows have a ticket to update this example and I've reverted it back to using the Alpha until then.

**Comment by Helcaraxan** [comment]
@majolo I am not happy merging (and releasing) this if the local flow is not supported.

This brings me to another point which is: why are we not removing the Alpha from the SDK?

**Comment by majolo** [comment]
Hey @Helcaraxan, so we're keeping the Alpha service available for one "cycle" to make the upgrade path smoother for users of the Platform SDK (they can get new features on other services without having to upgrade to Beta on the deployment service). 

With regards to blocking release of the cloud deployment APIs on feature parity with SpatialD, I agree that it isn't optimal but we also think it makes sense to release these now.

SpatialD is owned by Workflows and they have a ticket to do this work (https://improbableio.atlassian.net/browse/WF-1693) but it isn't a priority for them in Q4 so this will block releasing these APIs for a really long time.

Our customers can benefit from using these new APIs sooner and indeed the majority of our customers care about cloud with their room management systems. Also there are some important stability improvements that came out of a recent post mortem to prevent it from being easy to accidentally DOS the platform.

Users will still be able to interact with their local workflow using the Alpha until Workflows finish this supporting it.

I've spoken to Chris Cortes and he echoes these thoughts (that it's not optimal but worth doing for the benefit of our customers). There is also consensus on our team about this (Dima, Seb, BenY etc)

cc @sebzz

**Pull request merged**